### PR TITLE
fix: serverless failing multi-account/region tests

### PR DESCRIPTION
**Pull request by sannya-singal**
## Motivation
When using values other than `000000000000` for account ID or `us-east-1` for region, `serverless` tests should still create the consequent resources in these accounts and region. When we try to deploy the serverless setup using non-default region value in tests it throws `IndexError` or `AssertionError` since the setup wasn't getting deployed in the correct account.

## Changes
This PR: 
- passes `AWS_ACCESS_KEY_ID` as an environment variable to [`serverless-localstack`](https://github.com/localstack/serverless-localstack). 
- deletes(later undos), the `AWS_PROFILE` env variable. 
- replaces queue name with queue arn while fetching queue url in method `sqs_queue_url_for_arn`.
- passes `logs_client` in `get_lambda_log_events`. 

## Testing

The tests were failing previously when executed with a non-default account ID, set through environment variables (`TEST_AWS_ACCOUNT_ID=111111111111`, `TEST_AWS_ACCESS_KEY_ID=111111111111`, `TEST_AWS_REGION=us-west-1`). This PR fixes: 

`tests.aws.test_serverless.TestServerless.test_event_rules_deployed`
`tests.aws.test_serverless.TestServerless.test_dynamodb_stream_handler_deployed`
`tests.aws.test_serverless.TestServerless.test_kinesis_stream_handler_deployed`
`tests.aws.test_serverless.TestServerless.test_queue_handler_deployed`
`tests.aws.test_serverless.TestServerless.test_lambda_with_configs_deployed`
`tests.aws.test_serverless.TestServerless.test_apigateway_deployed`

### CODE REVIEW

File: localstack/constants.py
Comment (sannya-singal) [inline localstack/constants.py:160]: The following changes are just to showcase that the tests being addressed are now getting passed in the CI, these will be reverted before merge.

File: tests/aws/test_serverless.py
Comment (viren-nadkarni) [inline tests/aws/test_serverless.py:49]: Non-blocking: Is it possible to use the `TEST_AWS_ACCESS_KEY_ID` var here?

**Review by viren-nadkarni — APPROVED** [review]
A clean fix, LGTM! Good to merge once the changes in `localstack.constants` are reverted

**Pull request merged**
